### PR TITLE
Align log statements to read easier #406

### LINKED ISSUE #406: Align the log statements in log / mtrace
When logging padd out the names of each step with spaces so all the logs align

image

### PR DESCRIPTION
**Pull request by brendanheywood**
Resolves #406

### CODE REVIEW

**Comment by brendanheywood** [comment]
image

**Comment by keevan** [comment]
Oddly CI is throwing a hissy fit.

I was  wondering if we can simplify common bits, in particular when the engine says something vs when a step says it:
e.g. if the dataflow was called dataflow it might look something like
```
13:10:12.123 [dataflow][enrol user WS]: ............. Iteration 2: { something }
13:10:12.123 [dataflow][another step]: .............. Iteration 2: { something }
13:10:12.123 [dataflow][enrol user WS]: ............. Iteration 3: { something }
13:10:12.123 [dataflow]: ............................ Aborting, something went horribly wrong!
```

**Comment by brendanheywood** [comment]
> I was wondering if we can simplify common bits, in particular when the engine says something vs when a step says it:
e.g. if the dataflow was called dataflow it might look something like

yes but not now. I want to redo the whole way logging is done so that when we log we are dealing a bunch of different objects we can do things with and not just a lump of text. I also want to add logging levels. Right now I just want the reading pain to go away slightly. See https://github.com/catalyst/moodle-tool_dataflows/issues/381

File: classes/local/execution/logging_context.php [hunk 1]
```diff
@@ -62,11 +62,13 @@ public function log($message) {
...
-        $logstr = 'Engine "' . $this->engine->name . '"';
+        $logstr = "Engine '{$this->engine->name}'";
+        $strlen = min(20, strlen($this->enginestep->name));
+        $dots = str_repeat('.' , 25 - $strlen);
         if (!is_null($this->enginestep)) {
-            $logstr .= ', step "' . $this->enginestep->name . '"';
+            $logstr .= " step {$this->enginestep->name}{$dots}";
         }
-        $logstr .= ': ' . $message;
+        $logstr .= ' ' . $message;
```
Comment (keevan) [inline classes/local/execution/logging_context.php:71]: So it seems passing the `enginestep->name` (which is derived from a magic getter) into any function is the problem.

You can fix this immediate issue by putting it into a variable first
```suggestion
        $name = $this->engine->name;
        $logstr = "Engine '{$name}'";
        $strlen = min(20, strlen($name));
        $dots = str_repeat('.' , 25 - $strlen);
        if (!is_null($this->enginestep)) {
            $logstr .= " step {$name}{$dots}";
        }
        $logstr .= ' ' . $message;
```

File: classes/local/execution/logging_context.php [hunk 2]
```diff
@@ -62,11 +62,15 @@ public function log($message) {
...
         }
 
-        $logstr = 'Engine "' . $this->engine->name . '"';
+        $name = $this->engine->name;
+        $logstr = "Engine '{$name}'";
+        $strlen = min(20, strlen($name));
+        $dots = str_repeat('.' , 25 - $strlen);
         if (!is_null($this->enginestep)) {
-            $logstr .= ', step "' . $this->enginestep->name . '"';
+            $name = $this->engineistep->name;
```
Comment (keevan) [inline classes/local/execution/logging_context.php:70]: Slight typo here:
```suggestion
            $name = $this->enginestep->name;
```

Comment (brendanheywood) [reply classes/local/execution/logging_context.php:70]: my typing is haunted today

**Pull request merged**